python/example_code/bedrock-agent-runtime/flows/flow-conversation.py

```diff
@@ -0,0 +1,182 @@
+
+        # Check if more input us needed from user.
+        elif 'flowMultiTurnInputRequestEvent' in event:
+            input_required = event
+
+        # Print the model output.
+        elif 'flowOutputEvent' in event:
+            print(event['flowOutputEvent']['content']['document'])
+
+        # Log trace events.
+        elif 'flowTraceEvent' in event:
+            logger.info("Flow trace:  %s", event['flowTraceEvent'])
+
+    return {
+        "flow_status": flow_status,
+        "input_required": input_required,
+        "execution_id": execution_id
+    }
+
+
+def converse_with_flow(bedrock_agent_client, flow_id, flow_alias_id):
+    """
+    Run a conversation with the supplied flow.
+
+    Args:
+        bedrock_agent_client: Boto3 client for Amazon Bedrock agent runtime.
+        flow_id: The ID of the flow to run.
+        flow_alias_id: The alias ID of the flow.
+
+    """
+
+    flow_execution_id = None
+    finished = False
+
+    # Get the intial prompt from the user.
+    user_input = input("Enter input: ")
+
+    # Use prompt to create input data.
+    flow_input_data = {
+        "content": {
+            "document": user_input
+        },
+        "nodeName": "FlowInputNode",
+        "nodeOutputName": "document"
+    }
+
+    try:
+        while not finished:
+            # Invoke the flow until successfully finished.
+
+            result = invoke_flow(
+                bedrock_agent_client, flow_id, flow_alias_id, flow_input_data, flow_execution_id)
+
+            status = result['flow_status']
+            flow_execution_id = result['execution_id']
+            more_input = result['input_required']
+            if status == "INPUT_REQUIRED":
+                # The flow needs more information from the user.
+                logger.info("The flow %s requires more input", flow_id)
+                user_input = input(
+                    more_input['flowMultiTurnInputRequestEvent']['content']['document'] + ": ")
+                flow_input_data = {
+                    "content": {
+                        "document": user_input
+                    },
+                    "nodeName": more_input['flowMultiTurnInputRequestEvent']['nodeName'],
+                    "nodeInputName": "agentInputText"
+
+                }
+            elif status == "SUCCESS":
+                # The flow completed successfully.
+                finished = True
+                logger.info("The flow %s successfully completed.", flow_id)
+
+    except botocore.exceptions.ClientError as e:
+        print(f"Client error: {str(e)}")
+        logger.error("Client error: %s", {str(e)})
+
+    except Exception as e:
+        print(f"An error occurred: {str(e)}")
+        logger.error("An error occurred: %s", {str(e)})
+        logger.error("Error type: %s", {type(e)})
+
+
+def main():
+    """
+    Main entry point for the script.
+    """
+
+    # Replace these with your actual flow ID and flow alias ID.
+    FLOW_ID = 'YOUR_FLOW_ID'
```

Comment:
If these are required, could they be generated by a cloud formation stack or other automated means? Our automated test runner won't be able to populate the values.

Reply:
@rlhagerm would a unit test work, similar to https://github.com/awsdocs/aws-doc-sdk-examples/blob/main/python/example_code/bedrock-agent-runtime/test/test_bedrock_agent_runtime_wrapper.py

Reply:
Possibly you could get around the testing problem, but more generally we try to avoid forcing the user to do any required setup in the console. Instead we prefer to create resources either in a stack, CDK, or in the code (from our [coding standards](https://github.com/awsdocs/aws-doc-sdk-examples/wiki/general-code-example-standards)). Could you create the flow for the user as part of this example, and then use the IDs from that new flow? Of course you'd also have to clean it up after. But it would make this example meet the standard of "fully runnable".

Reply:
That could work (we do need to migrate the flow creation code examples). Typically, creating the flow would be a one time event for a client app. Would you want the example to create and tear down the flow resource each time the example is run?

Reply:
You could give the user the option to create/teardown (or not). I'll defer to your decision on it.

Reply:
Resolved via discussion and plans to add flow management in a separate update.